v1/plugin/stream_collector_proxy.go

```diff
@@ -61,16 +67,11 @@ func (p *StreamProxy) StreamMetrics(stream rpc.StreamCollector_StreamMetricsServ
 	// Metrics out of the plugin into snap.
 	outChan := make(chan []Metric)
 
-	err := p.plugin.StreamMetrics(inChan, outChan, errChan)
-	if err != nil {
-		return err
-	}
-
 	go p.metricSend(outChan, stream)
 	go p.errorSend(errChan, stream)
-	p.streamRecv(inChan, stream)
+	go p.streamRecv(inChan, stream)
 
-	return nil
+	return p.plugin.StreamMetrics(inChan, outChan, errChan)
```

Comment:
Consider making a separate pull request to submit changes related to the streaming area.